File: sdk/samples/all in on code/Visual Studio 2008/VBLinqExtension/LinqToCSVForm.vb
'****************************** Module Header ******************************\
' Module Name:    LinqToCSVForm.vb
' Project:        VBLinqExtension
' Copyright (c) Microsoft Corporation.
'
' This LinqToCSVForm creates a CSV document based on some of the 
' All-In-One Code Framework project information and uses the simple 
' LINQ to CSV library to read and display the CSV file contents.  
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
' All other rights reserved.
'
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'***************************************************************************/

#Region "Imports directives"
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports System.IO
#End Region

Public Class LinqToCSVForm

    Private Sub LinqToCSVForm_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        ' Create the CSV file
        CreateCSVFile()
    End Sub

    ' The file name of the CSV file
    Private filePath As String = "Projects.csv"

#Region "Load and Display CSV File"

    ' Use the simple LINQ to CSV library to load and display the CSV
    ' file contents.

    Private Sub btnLoad_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnLoad.Click
        Try
            ' Read the CSV file using StreamReader
            Using reader As New StreamReader(filePath)
                ' Use the extension method Lines of the StreamReader
                ' to load the contents into anonymous type objects

                Dim projects = From p In reader.Lines() _
                    Select New With {.ProjectName = p(0), .Language = p(1), .Description = p(2), .Owner = p(3)}

                ' Bind the data to DataGridView
                dgvLinqToCSV.DataSource = projects.ToList()
            End Using
        Catch ex As Exception
            MessageBox.Show(ex.ToString())
        End Try
    End Sub

#End Region


#Region "Create CSV File"

    ''' <summary>
    ''' This method creates a CSV file based on some of the All-In-One Code Framework project
    ''' information.
    ''' </summary>
    Private Sub CreateCSVFile()
        ' Create the CodoFx projects objects
        Dim projects As Project() = New Project(1) _
            {New Project() With {.ProjectName = "CSDllCOMServer", .Language = "C#", .Description = "An in-process COM server in C#", .Owner = "Jialiang Ge"}, _
             New Project() With {.ProjectName = "CSLinqToDataSets", .Language = "C#", .Description = "Use LINQ to DataSets in C#", .Owner = "Lingzhi Sun"}}

        ' Create the CSV file using StreamWriter
        Try
            Using writer As New StreamWriter(filePath)
                ' Write comment and data structure in CSV file
                writer.WriteLine("#All-In-One Code Framework Projects (format: Project Name, Program Language, Description, Owner)")

                ' Write the data
                For Each proj In projects
                    writer.WriteLine([String].Format(proj.ProjectName & "," & proj.Language & "," & proj.Description & "," & proj.Owner))
                Next
            End Using

            MessageBox.Show("Projects.csv created successfully!")
        Catch ex As Exception
            MessageBox.Show(ex.ToString())
        End Try
    End Sub

#End Region
End Class

#Region "The All-In-One Code Framework project class"

Public Class Project

    ' The project name
    Private _ProjectName As String
    Public Property ProjectName() As String
        Get
            Return (_ProjectName)
        End Get
        Set(ByVal value As String)
            _ProjectName = value
        End Set
    End Property

    ' The project programme language
    Private _Language As String
    Public Property Language() As String
        Get
            Return (_Language)
        End Get
        Set(ByVal value As String)
            _Language = value
        End Set
    End Property

    ' The project description
    Private _Description As String
    Public Property Description() As String
        Get
            Return (_Description)
        End Get
        Set(ByVal value As String)
            _Description = value
        End Set
    End Property

    ' The project owner
    Private _Owner As String
    Public Property Owner() As String
        Get
            Return (_Owner)
        End Get
        Set(ByVal value As String)
            _Owner = value
        End Set
    End Property
End Class
#End Region

File: sdk/samples/all in on code/Visual Studio 2008/CSLinqExtension/LinqToCSVForm.cs
/****************************** Module Header ******************************\
* Module Name:    LinqToCSVForm.cs
* Project:        CSLinqExtension
* Copyright (c) Microsoft Corporation.
*
* This LinqToCSVForm creates a CSV document based on some of the 
* All-In-One Code Framework project information and uses the simple 
* LINQ to CSV library to read and display the CSV file contents.  
* 
* This source is subject to the Microsoft Public License.
* See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
* All other rights reserved.
*
* THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
* EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
* WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\***************************************************************************/

#region Using directives
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.IO;
#endregion


namespace CSLinqExtension
{
    public partial class LinqToCSVForm : Form
    {
        public LinqToCSVForm()
        {
            InitializeComponent();

            // Create the CSV file
            CreateCSVFile();
        }

        // The file name of the CSV file
        private string filePath = "Projects.csv";


        #region Load and Display CSV File

        // Use the simple LINQ to CSV library to load and display the CSV
        // file contents.
        private void btnLoad_Click(object sender, EventArgs e)
        {
            try
            {
                // Read the CSV file using StreamReader
                using (StreamReader reader = new StreamReader(filePath))
                {
                    // Use the extension method Lines of the StreamReader
                    // to load the contents into anonymous type objects
                    var projects = from p in reader.Lines()
                                   select new { ProjectName = p[0], Language = p[1], Description = p[2], Owner = p[3] };

                    // Bind the data to DataGridView
                    dgvLinqToCSV.DataSource = projects.ToList();
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.ToString());
            }
        }

        #endregion


        #region Create CSV File

        /// <summary>
        /// This method creates a CSV file based on some of the All-In-One Code Framework project
        /// information.
        /// </summary>
        private void CreateCSVFile()
        {
            // Create the CodoFx projects objects
            Project[] projects = new Project[2] 
            {
                new Project() { ProjectName = "CSDllCOMServer",Language = "C#", Description = "An in-process COM server in C#", Owner = "Jialiang Ge" },
                new Project() { ProjectName = "CSLinqToDataSets", Language = "C#", Description = "Use LINQ to DataSets in C#", Owner = "Lingzhi Sun" },
            };

            // Create the CSV file using StreamWriter
            try
            {
                using (StreamWriter writer = new StreamWriter(filePath))
                {
                    // Write comment and data structure in CSV file
                    writer.WriteLine("#All-In-One Code Framework Projects (format: Project Name, Program Language, Description, Owner)");

                    // Write the data
                    foreach (var proj in projects)
                    {
                        writer.WriteLine(String.Format("{0},{1},{2},{3}", proj.ProjectName, proj.Language, proj.Description, proj.Owner));
                    }
                }

                MessageBox.Show("Projects.csv created successfully!");
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.ToString());
            }
        }

        #endregion
    }


    #region The All-In-One Code Framework project class
    public class Project
    {
        // The project name
        public string ProjectName { get; set; }

        // The project programme language
        public string Language { get; set; }

        // The project description
        public string Description { get; set; }

        // The project owner
        public string Owner { get; set; }
    }
    #endregion
}
